custom/icds/management/commands/find_cases_with_no_delivery.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import csv
+
+from django.core.management import BaseCommand
+
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.form_processor.models import CommCareCaseSQL
+from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+
+from dimagi.utils.chunked import chunked
+
+
+class Command(BaseCommand):
+    """https://manage.dimagi.com/default.asp?265190
+
+    Some ccs_record cases have evidence of a delivery occurring, but do not
+    have an associated delivery form. The implications on the case are that an
+    add property is not set and the schedule phase is 2.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'csv_file',
+            help="File path for csv file",
+        )
+
+    def handle(self, csv_file, **options):
+        self.domain = 'icds-cas'
+        self.case_accessor = CaseAccessors(self.domain)
+        with open(csv_file, "w") as csv_file:
+            field_names = [
+                'case_id', 'owner_id', 'modified_on', 'server_modified_on',
+                'add', 'edd', 'ccs_phase', 'num_pnc_visits', 'current_schedule_phase'
+            ]
+
+            csv_writer = csv.DictWriter(csv_file, field_names, extrasaction='ignore')
+            csv_writer.writeheader()
+
+            for ccs_case in self._get_cases():
+                properties = ccs_case.case_json
+
+                if 'add' in properties:
+                    continue
+
+                if properties.get('current_schedule_phase') != 2:
+                    continue
+
+                properties.update({
```

Comment:
this will update the case object's actual case_json since it's still a reference to the same field. it doesn't matter because the case is not being saved but it may have been better to copy the json rather than use the same reference